migration/src/spreadsheets.rs

```diff
@@ -130,6 +132,68 @@ impl SheetResult {
         })
     }
 
+    /// Parse this sheet as the collection index. Updates a Collection 
+    /// which contains two Chapter Groups, each of which 
+    /// has a list of associated Chapters.
+    pub fn into_collection_index(self,
+        self_title: &String, 
+        self_wordpress_menu_id: &i64) -> Result<Collection> {
+        let mut chapters = Vec::new();
+        let mut self_intro_chapters = Vec::new();
+        for row in self.values.into_iter().skip(2){
+            if row[0].is_empty(){
+                self_intro_chapters = chapters.clone();
```

Comment:
You're doing a ton of cloning in this function as well, which you shouldn't need to do. You can do something like this instead:

```rs
let row = row.into_iter();
let first_value = row.next().ok_or_else(|| anyhow!("Missing first value"))?;
let second_value = row.next().ok_or_else(|| anyhow!("Missing second value"))?;
```
It's not the shortest but it's more idiomatic.